### PR TITLE
Add subworkflow to get sequences

### PR DESCRIPTION
**Pull request by itrujnara**
Add a subworkflow that obtains ortholog sequences from databases. It's much more robust than the current module.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests! _Not yet, we'll get there_
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).

### CODE REVIEW

**Comment by itrujnara** [comment]
@nf-core-bot fix linting

File: subworkflows/local/get_sequences.nf [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    FETCH_OMA_SEQUENCES(ch_idfiles.oma.join(ch_query_fasta))
+    ch_fasta    = ch_fasta.mix(FETCH_OMA_SEQUENCES.out.fasta)
+    ch_hits     = ch_hits.mix(FETCH_OMA_SEQUENCES.out.hits)
+    ch_misses   = ch_misses.mix(FETCH_OMA_SEQUENCES.out.misses)
+    ch_versions = ch_versions.mix(FETCH_OMA_SEQUENCES.out.versions)
+
+    ch_fasta_grouped = ch_fasta.groupTuple()
+    ch_hits_grouped = ch_hits.groupTuple()
+    ch_misses_grouped = ch_misses.groupTuple()
```
Comment (JoseEspinosa) [inline subworkflows/local/get_sequences.nf]:
```suggestion
    ch_fasta_grouped  = ch_fasta.groupTuple()
    ch_hits_grouped   = ch_hits.groupTuple()
    ch_misses_grouped = ch_misses.groupTuple()
```

File: subworkflows/local/get_sequences.nf [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    ch_ids
+    ch_query_fasta
+
+    main:
+    ch_versions = Channel.empty()
+
+    SPLIT_ID_FORMAT(ch_ids)
+    ch_versions = ch_versions.mix(SPLIT_ID_FORMAT.out.versions)
+
+    ch_idfiles = SPLIT_ID_FORMAT.out.ids_split.transpose().branch {
```
Comment (JoseEspinosa) [inline subworkflows/local/get_sequences.nf]: For consistency with the rest of the naming I will use `ch_id_files` but it is just a suggestion, if you prefer it this way keep it
```suggestion
    ch_id_files = SPLIT_ID_FORMAT.out.ids_split.transpose().branch {
```

File: subworkflows/local/get_sequences.nf [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    ch_fasta    = ch_fasta.mix(FETCH_UNIPROT_SEQUENCES.out.fasta)
+    ch_hits     = ch_hits.mix(FETCH_UNIPROT_SEQUENCES.out.hits)
+    ch_misses   = ch_misses.mix(FETCH_UNIPROT_SEQUENCES.out.misses)
+    ch_versions = ch_versions.mix(FETCH_UNIPROT_SEQUENCES.out.versions)
+
+    FETCH_ENSEMBL_IDMAP()
+    ch_versions = ch_versions.mix(FETCH_ENSEMBL_IDMAP.out.versions)
+
+    FETCH_ENSEMBL_SEQUENCES(
+        ch_idfiles.ensembl.join(ch_query_fasta),
```
Comment (JoseEspinosa) [inline subworkflows/local/get_sequences.nf]:
```suggestion
        ch_id_files.ensembl.join(ch_query_fasta),
```

File: conf/modules.config [hunk 4]
```diff
@@ -154,16 +183,6 @@ process {
     // Sequence alignment
     // ----------------------
 
-    withName: 'FETCH_SEQUENCES_ONLINE' {
```
Comment (JoseEspinosa) [inline conf/modules.config:157]: This is just gone and not replaced by `FETCH_OMA_SEQUENCES` ?

File: modules/local/convert_fasta.nf [hunk 5]
```diff
@@ -2,10 +2,11 @@ process CONVERT_FASTA {
     tag "$input_file"
     label 'process_single'
 
-    conda "conda-forge::python=3.11.0 conda-forge::biopython=1.83.0 conda-forge::requests=2.31.0"
+    // THINGS ARE WRONG HERE, FIXME WHEN SEQERA CONTAINTAINERS START TO COOPERATE
```
Comment (JoseEspinosa) [inline modules/local/convert_fasta.nf]: What does this comment mean?

File: modules/local/convert_fasta.nf [hunk 6]
```diff
@@ -2,10 +2,11 @@ process CONVERT_FASTA {
...
+    conda "conda-forge::python=3.12.0 conda-forge::biopython=1.84.0 conda-forge::requests=2.32.3"
```
Comment (JoseEspinosa) [inline modules/local/convert_fasta.nf:5]: At some point, for sure not for this PR, maybe think of adopting the structure of the nf-core modules, with an `envinroment.yml` and maybe tests?

File: modules/local/fetch_refseq_sequences.nf [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        'biocontainers/mulled-v2-bc54124b36864a4af42a9db48b90a404b5869e7e:5258b8e5ba20587b7cbf3e942e973af5045a1e59-0' }"
+
+    input:
+    tuple val(meta), path(ids), path(query_fasta)
+
+    output:
+    tuple val(meta), path("*_refseq_sequences.fa")  , emit: fasta
+    tuple val(meta), path("*_refseq_seq_hits.txt")  , emit: hits
+    tuple val(meta), path("*_refseq_seq_misses.txt"), emit: misses
+    path "versions.yml"                          , emit: versions
```
Comment (JoseEspinosa) [inline modules/local/fetch_refseq_sequences.nf]:
```suggestion
    tuple val(meta), path("*_refseq_seq_misses.txt"), emit: misses
    path "versions.yml"                             , emit: versions
```

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Very nice job! 🚀  
Some minor comments from my side

File: conf/modules.config (hunk 4 above)
Comment (itrujnara) [reply conf/modules.config:157]: It is indeed replaced

File: modules/local/convert_fasta.nf (hunk 6 above)
Comment (itrujnara) [reply modules/local/convert_fasta.nf:5]: Perhaps

**Review by JoseEspinosa — APPROVED** [review]
Thanks! LGTM 🚀

**Pull request merged**